common/scripted_effects/wc_artifacts_scripted_effects.txt

```diff
@@ -51,9 +63,141 @@ set_prefix_and_suffix_flags_to_weapon_artifact_effect = {
+		ring_4_learning				= { set_name="ring_4_learning"			}
+
+		ring_5_diplomacy			= { set_name="ring_5_diplomacy"			}
+		ring_5_martial				= { set_name="ring_5_martial"			}
+		ring_5_stewardship			= { set_name="ring_5_stewardship"		}
+		ring_5_intrigue				= { set_name="ring_5_intrigue"			}
+		ring_5_learning				= { set_name="ring_5_learning"			}
+
+		## Weapons
+		# Two-handed Swords
+		sword_twohanded_1			= { set_name="sword_twohanded_1"		}
+		sword_twohanded_2			= { set_name="sword_twohanded_2"		}
+		sword_twohanded_3			= { set_name="sword_twohanded_3"		}
+		sword_twohanded_4			= { set_name="sword_twohanded_4"		}
+		sword_twohanded_5			= { set_name="sword_twohanded_5"		}
+		# One-handed Swords
+		sword_onehanded_1			= { set_name="sword_onehanded_1"		}
+		sword_onehanded_2			= { set_name="sword_onehanded_2"		}
+		sword_onehanded_3			= { set_name="sword_onehanded_3"		}
+		sword_onehanded_4			= { set_name="sword_onehanded_4"		}
+		sword_onehanded_5			= { set_name="sword_onehanded_5"		}
+		# Two-handed Axes
+		axe_twohanded_1				= { set_name="axe_twohanded_1"			}
+		axe_twohanded_2				= { set_name="axe_twohanded_2"			}
+		axe_twohanded_3				= { set_name="axe_twohanded_3"			}
+		axe_twohanded_4				= { set_name="axe_twohanded_4"			}
+		axe_twohanded_5				= { set_name="axe_twohanded_5"			}
+		# One-handed Axes
+		axe_onehanded_1				= { set_name="axe_onehanded_1"			}
+		axe_onehanded_2				= { set_name="axe_onehanded_2"			}
+		axe_onehanded_3				= { set_name="axe_onehanded_3"			}
+		axe_onehanded_4				= { set_name="axe_onehanded_4"			}
+		axe_onehanded_5				= { set_name="axe_onehanded_5"			}
+		# Two-handed Maces
+		mace_twohanded_1			= { set_name="mace_twohanded_1"			}
+		mace_twohanded_2			= { set_name="mace_twohanded_2"			}
+		mace_twohanded_3			= { set_name="mace_twohanded_3"			}
+		mace_twohanded_4			= { set_name="mace_twohanded_4"			}
+		mace_twohanded_5			= { set_name="mace_twohanded_5"			}
+		# One-handed Maces
+		mace_onehanded_1			= { set_name="mace_onehanded_1"			}
+		mace_onehanded_2			= { set_name="mace_onehanded_2"			}
+		mace_onehanded_3			= { set_name="mace_onehanded_3"			}
+		mace_onehanded_4			= { set_name="mace_onehanded_4"			}
+		mace_onehanded_5			= { set_name="mace_onehanded_5"			}
+	}
+
+	clear_flags_with_prefix = temp_artifact_
+}
+
+give_generic_artifact_random_name_and_origin_effect = {
+	if = {
+		limit = { quality == 1 }
+		set_description = "artifact_common_origin"
+	}
+	else_if = {
+		limit = { quality == 2 }
+		set_description = "artifact_uncommon_origin"
+	}
+	else_if = {
+		limit = { quality == 3 }
+		set_description = "artifact_rare_origin"
+	}
+	else_if = {
+		limit = { quality == 4 }
+		set_description = "artifact_epic_origin"
+	}
+	else = { set_description = "artifact_legendary_origin" }
+	give_generic_artifact_random_name_effect = yes
+}
+
 finish_artifact_creation_effect = {
 	set_creation_date = 1.1.1
-	clear_flags_with_prefix = temp_artifact_
 	log = "Added generic artifact: [This.GetName]"
 }
```

Comment:
Looks like `finish_artifact_creation_effect` can be moved inside `give_generic_artifact_random_name_and_origin_effect`.

Reply:
`set_creation_date = 1.1.1` effect is applicable for artifacts that were crafted in unknown times and `log` effects that is only needed for artifacts that spawns before game start... So, not all artifacts need both effects.
For now, I only reworked naming effects, artifact spawning effects can be reworked later.